pkg/controller/certificates/trigger/policies/gatherer_test.go

```diff
@@ -0,0 +1,185 @@
+/*
+Copyright 2020 The Jetstack cert-manager contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package policies
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	cmapi "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+func TestDataForCertificate(t *testing.T) {
+	tests := []struct {
+		name           string
+		givenGetSecret func(t *testing.T) func(string) (*v1.Secret, error)
+		givenListReq   func(t *testing.T) func(labels.Selector) ([]*cmapi.CertificateRequest, error)
+		givenCrt       *cmapi.Certificate
+		want           Input
+		wantErr        string
+	}{
+		{
+			name:           "the returned secret should stay nil when it is not found",
+			givenCrt:       &cmapi.Certificate{Spec: cmapi.CertificateSpec{SecretName: "secret-1"}, ObjectMeta: metav1.ObjectMeta{Name: "a"}},
+			givenGetSecret: mockGetSecret("secret-1", nil, apierrors.NewNotFound(cmapi.Resource("Secret"), "secret-1")),
+			givenListReq:   mockListCR([]*cmapi.CertificateRequest{}, nil),
+			want:           Input{Secret: nil},
+		},
+		{
+			name:           "should return an error when getsecret returns an unexpected error",
+			givenCrt:       &cmapi.Certificate{Spec: cmapi.CertificateSpec{SecretName: "secret-1"}, ObjectMeta: metav1.ObjectMeta{Name: "a"}},
+			givenGetSecret: mockGetSecret("secret-1", nil, fmt.Errorf("unexpected")),
+			givenListReq:   mockListCR([]*cmapi.CertificateRequest{}, nil),
+			wantErr:        "unkexpected",
```

Comment:
I hope we don't throw this error

Reply:
Oops, apparently I wasn't even testing that `wantErr` matched the returned error... oops